src/presets/deno/preset.ts

```diff
@@ -43,109 +38,28 @@ const denoServer = defineNitroPreset(
-        setImmediate: ["node:timers", "setImmediate"],
-        clearImmediate: ["node:timers", "clearImmediate"],
-        // process: ["node:process", "default"],
-      },
+      preview: "deno task start",
     },
     rollupConfig: {
+      external: (id) =>
+        id.startsWith("https://") ||
+        id.startsWith("node:") ||
+        builtinModules.includes(id),
       output: {
         hoistTransitiveImports: false,
       },
-      plugins: [
-        {
-          name: "rollup-plugin-node-deno",
-          resolveId(id) {
-            id = id.replace("node:", "");
-            if (builtinModules.includes(id)) {
-              return {
-                id: `node:${id}`,
-                moduleSideEffects: false,
-                external: true,
-              };
-            }
-            if (isHTTPImport(id)) {
-              return {
-                id,
-                external: true,
-              };
-            }
-          },
-          renderChunk(code) {
-            const s = new MagicString(code);
-            const imports = findStaticImports(code);
-            for (const i of imports) {
-              if (
-                !i.specifier.startsWith(".") &&
-                !isAbsolute(i.specifier) &&
-                !isHTTPImport(i.specifier) &&
-                !i.specifier.startsWith("npm:")
-              ) {
-                const specifier = i.specifier.replace("node:", "");
-                s.replace(
-                  i.code,
-                  i.code.replace(
-                    new RegExp(`(?<quote>['"])${i.specifier}\\k<quote>`),
-                    JSON.stringify(
-                      builtinModules.includes(specifier)
-                        ? "node:" + specifier
-                        : "npm:" + specifier
-                    )
-                  )
-                );
-              }
-            }
-            if (s.hasChanged()) {
-              return {
-                code: s.toString(),
-                map: s.generateMap({ includeContent: true }),
-              };
-            }
-          },
-        },
-        {
-          name: "inject-process",
-          renderChunk: {
-            order: "post",
-            handler(code, chunk) {
-              if (
-                !chunk.isEntry &&
-                (!ImportMetaRe.test(code) || code.includes("ROLLUP_NO_REPLACE"))
-              ) {
-                return;
-              }
-
-              const s = new MagicString(code);
-              s.prepend(
-                "import __process__ from 'node:process';globalThis.process=globalThis.process||__process__;"
-              );
-
-              return {
-                code: s.toString(),
-                map: s.generateMap({ includeContent: true }),
-              };
-            },
-          },
-        },
-      ],
+    },
+    alias: {
+      // Deno supports `fetch()` natively everywhere
+      "node-fetch-native": "unenv/runtime/mock/empty",
```

Comment:
Note that `node-fetch-native` main subpath is **not** a polyfill. It uses [export conditions](https://github.com/unjs/node-fetch-native/blob/main/package.json#L12) and in case of deno, it is using native fetch.

This line should be avoided, because unenv redirects legacy packages in dependency tree (like `node-fetch` usages) to node-fetch-native (as stub)

Reply:
Without this line it always includes the `fetch` polyfill in the generated server code. Maybe something else is not working correctly.

Reply:
Nice, didn't know about the playground. That makes it a lot easier to work on presets. I stand corrected, looks like I was looking at an old build. Can confirm that it's not present. Thanks for taking a look 👍

Reply:
Removed alias in d0e9c73686d6adaa7c753ae971065e29703e263a (screenshot above is from the state without using alias)